gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/organizations/environments/domains/ProtectedResourcesResource.java

```diff
@@ -73,12 +73,61 @@ public void createProtectedResource(
 
         checkAnyPermission(organizationId, environmentId, domain, Permission.PROTECTED_RESOURCE, Acl.CREATE)
                 .andThen(checkDomainExists(domain)
-                        .flatMap(existingDomain -> protectedResourceService.create(existingDomain, authenticatedUser, newProtectedResource)
+                        .flatMap(existingDomain -> service.create(existingDomain, authenticatedUser, newProtectedResource)
                                 .map(protectedResource -> Response
                                         .created(URI.create("/organizations/" + organizationId + "/environments/" + environmentId + "/domains/" + domain + "/protected-resources/" + protectedResource.getId()))
                                         .entity(protectedResource)
                                         .build())))
                 .subscribe(response::resume, response::resume);
     }
 
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(
+            operationId = "listProtectedResources",
+            summary = "List registered protected resources for a security domain",
+            description = "User must have the PROTECTED_RESOURCE[LIST] permission on the specified domain, environment or organization " +
+                    "AND either PROTECTED_RESOURCE[READ] permission on each domain's application " +
+                    "or PROTECTED_RESOURCE[READ] permission on the specified domain " +
+                    "or PROTECTED_RESOURCE[READ] permission on the specified environment " +
+                    "or PROTECTED_RESOURCE[READ] permission on the specified organization. " +
+                    "Each returned application is filtered and contains only basic information such as id, name, description and isEnabled.")
+    @ApiResponses({
+            @ApiResponse(responseCode = "200",
+                    description = "List registered protected resources for a security domain",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = ProtectedResourcesResource.ProtectedResourcePage.class))
+            ),
+            @ApiResponse(responseCode = "500", description = "Internal server error")})
+    public void list(
+            @PathParam("organizationId") String organizationId,
+            @PathParam("environmentId") String environmentId,
+            @PathParam("domain") String domain,
+            @QueryParam("type") String type,
+            @QueryParam("page") @DefaultValue("0") int page,
+            @QueryParam("size") @DefaultValue("50") int size,
```

Comment:
suggestion: I would add also a sort parameter to specify the field on which we want to sort (if missing, sort by internalId as currently), this will help to have a useful pagination mechanism on the UI as currently we only sort within the current page... wdyt?